x-pack/plugins/security_solution/public/common/containers/sourcerer/get_sourcerer_data_view.ts

```diff
@@ -7,27 +7,51 @@
 
 import type { DataViewsContract } from '@kbn/data-views-plugin/common';
 import { ensurePatternFormat } from '../../../../common/utils/sourcerer';
+import type { SourcererDataView } from '../../store/sourcerer/model';
+import { getDataViewStateFromIndexFields } from '../source/use_data_view';
 
 export const getSourcererDataView = async (
   dataViewId: string,
   dataViewsService: DataViewsContract,
   refreshFields = false
-) => {
+): Promise<SourcererDataView> => {
   const dataViewData = await dataViewsService.get(dataViewId, true, refreshFields);
   const defaultPatternsList = ensurePatternFormat(dataViewData.getIndexPattern().split(','));
-  const patternList = defaultPatternsList.reduce((res: string[], pattern) => {
-    if (dataViewData.matchedIndices.find((q) => q.includes(pattern.replaceAll('*', '')))) {
-      res.push(pattern);
+  let patternList = [];
+
+  try {
+    for (const pattern of defaultPatternsList) {
```

Comment:
nit -> Maybe we can parallelized this way to gain some speed 🚤 🐎 🚀 
```suggestion
    const validPatternist = await Promise.allSettled(
      defaultPatternsList.map((pattern) =>
        dataViewsService
          .getFieldsForWildcard({
            type: dataViewData.type,
            rollupIndex: dataViewData?.typeMeta?.params?.rollup_index,
            allowNoIndex: false,
            pattern,
          })
          .then(() => true)
          .catch(() => false)
      )
    );
    patternList = defaultPatternsList.filter((pattern, idx) => validPatternist[idx]);
```